core/src/test/java/com/sequenceiq/cloudbreak/core/flow2/cluster/ccm/upgrade/CcmUpgradeActionsTest.java

```diff
@@ -0,0 +1,282 @@
+import com.sequenceiq.cloudbreak.common.service.Clock;
+import com.sequenceiq.cloudbreak.common.service.TransactionMetricsService;
+import com.sequenceiq.cloudbreak.common.service.TransactionService;
+import com.sequenceiq.cloudbreak.core.flow2.CloudbreakFlowInformation;
+import com.sequenceiq.cloudbreak.core.flow2.StackStatusFinalizer;
+import com.sequenceiq.cloudbreak.core.flow2.service.CbEventParameterFactory;
+import com.sequenceiq.cloudbreak.core.flow2.service.ReactorNotifier;
+import com.sequenceiq.cloudbreak.core.flow2.stack.CloudbreakFlowMessageService;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.StackStatus;
+import com.sequenceiq.cloudbreak.domain.stack.cluster.Cluster;
+import com.sequenceiq.cloudbreak.domain.view.ClusterView;
+import com.sequenceiq.cloudbreak.domain.view.StackView;
+import com.sequenceiq.cloudbreak.logger.concurrent.MDCCleanerThreadPoolExecutor;
+import com.sequenceiq.cloudbreak.reactor.api.event.StackEvent;
+import com.sequenceiq.cloudbreak.reactor.handler.cluster.upgrade.ccm.CcmRemoveAutoSshHandler;
+import com.sequenceiq.cloudbreak.reactor.handler.cluster.upgrade.ccm.CcmReregisterClusterToClusterProxyHandler;
+import com.sequenceiq.cloudbreak.reactor.handler.cluster.upgrade.ccm.CcmUnregisterHostsHandler;
+import com.sequenceiq.cloudbreak.reactor.handler.cluster.upgrade.ccm.CcmUpgradePreparationHandler;
+import com.sequenceiq.cloudbreak.service.StackUpdater;
+import com.sequenceiq.cloudbreak.service.cluster.ClusterService;
+import com.sequenceiq.cloudbreak.service.publicendpoint.ClusterPublicEndpointManagementService;
+import com.sequenceiq.cloudbreak.service.secret.service.SecretService;
+import com.sequenceiq.cloudbreak.service.stack.InstanceMetaDataService;
+import com.sequenceiq.cloudbreak.service.stack.StackService;
+import com.sequenceiq.cloudbreak.util.StackUtil;
+import com.sequenceiq.flow.api.model.FlowIdentifier;
+import com.sequenceiq.flow.core.FlowRegister;
+import com.sequenceiq.flow.core.stats.FlowOperationStatisticsService;
+import com.sequenceiq.flow.domain.FlowLog;
+import com.sequenceiq.flow.reactor.eventbus.ConsumerCheckerEventBus;
+import com.sequenceiq.flow.reactor.handler.ConsumerNotFoundHandler;
+import com.sequenceiq.flow.repository.FlowChainLogRepository;
+import com.sequenceiq.flow.repository.FlowLogRepository;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.FreeIpaV1Endpoint;
+import com.sequenceiq.notification.NotificationService;
+
+import io.jaegertracing.internal.JaegerTracer;
+import io.jaegertracing.internal.reporters.InMemoryReporter;
+import io.jaegertracing.internal.samplers.ConstSampler;
+import io.jaegertracing.spi.Reporter;
+import io.jaegertracing.spi.Sampler;
+import reactor.Environment;
+import reactor.bus.EventBus;
+import reactor.bus.spec.EventBusSpec;
+import reactor.core.dispatch.MpscDispatcher;
+
+@ExtendWith(SpringExtension.class)
+class CcmUpgradeActionsTest {
```

Comment:
Perhaps insert `Integration` (or `Component`) before `Test`, as this is much more than a typical UT. :smile:

Reply:
ok